examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAliveFramework.scala

```diff
@@ -95,64 +95,111 @@ class KeepAliveFramework(settings: KeepAliveFrameWorkSettings, authorization: Op
       +-----------------------+
 
     */
-  source
+  val end = source
     .via(keepAliveWatcher)
     .prepend(Source(specsSnapshot.map { spec =>
       LaunchPod(spec.id, spec.runSpec)
     }))
-    .to(sink)
-    .run()
+    .runWith(sink)
 
   // We let the framework run "forever"
-  io.StdIn.readLine("Keep alive framework is started")
-
+  val result = Await.result(end, Duration.Inf)
```

Comment:
I was hoping that the stream would finish with an error when the client cannot connect but it keeps running. `asSourceAndSink` seems to be broken or I am missing something obvious. @timcharper, how can we let the framework crash on a connection error etc?

Reply:
Hmm, we have unit tests specifically for this. Maybe something else is going on?